src/app/shared/top-panel/user-panel/notifications-menu/notifications-menu.component.html

```diff
@@ -0,0 +1,200 @@
+        <i class="fa fa-bell fa-stack-2x"
+           [class.text-primary]="(unreadNotifications$ | async)?.length > 0"
+           [class.text-secondary]="!(unreadNotifications$ | async) || (unreadNotifications$ | async).length === 0"
+        ></i>
+        <span class="fa-stack-1x" *ngIf="(unreadNotifications$ | async)?.length > 0">
+          <span class="notifications-count">{{(unreadNotifications$ | async).length}}</span>
+        </span>
+      </span>
+    </a>
+
+    <div class="dropdown-menu dropdown-menu-right pre-scrollable"
+         aria-labelledby="notificationsDropdownLink">
+
+      <form class="px-2 text-right">
+        <span class="small text-muted mr-1 align-text-top">Show only unread</span>
+        <label class="switch">
+          <input type="checkbox" [checked]="unreadOnly"  (click)="toggleUnreadFilter($event)">
+          <span class="slider round" (click)="$event.stopPropagation()"></span>
+        </label>
+      </form>
+
+      <a *ngFor="let notification of ((unreadOnly ? unreadNotifications$ : notifications$) | async);"
+         class="dropdown-item cursor-default">
+        <span class="notification-title clearfix">
+          <span class="status small text-muted float-left">{{notification.createdAt | date:'shortDate' }}</span>
+          <span class="status float-right"
+                [class.read]="notification.read"
+                [class.unread]="!notification.read">
+            <span class="align-text-top small text-muted">
+              <span *ngIf="!notification.firstReadAt" i18n="@@userNotifications.new">new</span>
+              <span *ngIf="notification.read && notification.firstReadAt" i18n="@@userNotifications.readOn">read on</span>
+              <span *ngIf="!notification.read && notification.firstReadAt" i18n="@@userNotifications.originallyReadOn">originally read on</span>
+              <span *ngIf="notification.firstReadAt">{{' ' + (notification.firstReadAt | date:'shortDate')}}</span>
+            </span>
+            <i *ngIf="notification.read"
+               (click)="setNotificationReadStatus($event, notification, false)"
+               class="fa fa-eye text-success ml-1 cursor-pointer"
+               title="Mark as unread"
+               title.i18n="@@userNotifications.mark.unread"></i>
+            <i *ngIf="!notification.read"
+               (click)="setNotificationReadStatus($event, notification, true)"
+               class="fa fa-eye-slash text-success ml-1 cursor-pointer"
+               title="Mark as read"
+               title.i18n="@@userNotifications.mark.read"></i>
+          </span>
+        </span>
+        <span class="notification-content read text-notice cursor-pointer"
+              [class.read]="notification.read"
+              [class.text-notice]="notification.read"
+              [class.unread]="!notification.read">
+          <ng-container *ngTemplateOutlet="getTemplate(notification.type); context: { $implicit: notification }"></ng-container>
```

Comment:
This component is way too big. It can be split into at least 7 different components.

For example this very unusual switch can be done in a separate component using Angular's [ngSwitch](https://angular.io/api/common/NgSwitch) directive. And each notification type can have its own component.

Reply:
This was partially laziness, partially saving of time. Maybe some time in the future. I think the component for individual types can be more generic. I suggest to resolve this once we add TaskUserNotification type.